[Location] kakao검색 결과 기반 google map 마커 띄우기

jinju9513 opened the pull request:
### 🚀: 개요
- route 정보에 맞게 지도 마커 생성
- 앱바에 여행 정보(제목/날짜) 생성
- 마커 및 바텀시트 연동

### 🚀: 작업 내용
- Firebase에서 불러온 장소 데이터를 기반으로 지도에 마커 표시
- 장소 카드(PageView)와 마커 간 연동 구현 (카드 클릭 시 지도 이동, 마커 클릭 시 카드 스크롤)
- 첫 진입 시 선택된 장소 기준으로 카메라 자동 이동
- 마커 탭 시 해당 장소의 카드로 PageView 애니메이션 이동
- 카드 넘기면 지도도 함께 이동하도록 처리
- 뒤로가기 → 재진입 시 앱 종료되던 문제 해결
- plans 문서에서 불러온 title 및 date를 앱바에 표시

### 📸: 실행 화면
![Screenshot_1749467536](https://github.com/user-attachments/assets/b6dce9d7-ba3a-4900-be4a-126d53f0b966)

### 🚀: 조정 파일
- map_provider.dart
- map_view_model
- place_carousel.dart
- map_page.dart
- AppDelegate.swift
- schedule_app_bar.dart
- map_display.dart
- place_card.dart
- map_repository.dart
- map_state.dart

### 개발 결과
- 지도의 마커 - 카드 연동
- 여행 제목과 날짜가 앱바에 노출
- 앱 종료/오류 없이 안정적으로 페이지 진입 및 이동 가능
- TabController 관련 예외 없이 탭 간 카메라 이동 및 상태 전환 작동

### issue
Closes #56

jinju9513 committed:
[Location] kakao검색 결과 기반 google map 마커 띄우기
Fixes #56
feat: MapState 모델 정의 및 copyWith 메서드 추가

jinju9513 committed:
feat: mapViewModelProvider 및 mapRepositoryProvider 설정

jinju9513 committed:
feat: Firestore에서 플랜 및 경로 데이터를 불러오는 mapRepository 구현

jinju9513 committed:
feat: MapRepository 생성 및 Firestore 연동 로직 분리

jinju9513 committed:
feat: MapViewModel 구현 및 지도 탭/카드 상태 관리 기능 추가

jinju9513 committed:
feat: 여행 플랜 데이터 로드 및 마커 생성 기능 구현, UI 위젯 분리 (MapPage)

jinju9513 committed:
feat: 테스트용 MapPage 라우팅 설정 및 화면 연결 확인

jinju9513 committed:
feat: iOS AppDelegate에 Google Maps API 키 등록 로직 추가

jinju9513 committed:
fix: MapPage 재진입 시 dispose된 controller 재사용 오류 해결

ghghtls commented on lib/views/location/widgets/place_carousel.dart:
```diff
@@ -0,0 +1,69 @@
+            final lat = double.tryParse(place['lat'] ?? '') ??
+                double.tryParse(place['latitude'] ?? '');
+            final lng = double.tryParse(place['lng'] ?? '') ??
+                double.tryParse(place['longitude'] ?? '');
```
LatLng? parseLatLng(Map<String, dynamic> data) {
  final lat = double.tryParse(data['lat'] ?? '') ?? double.tryParse(data['latitude'] ?? '');
  final lng = double.tryParse(data['lng'] ?? '') ?? double.tryParse(data['longitude'] ?? '');
  if (lat == null || lng == null) return null;
  return LatLng(lat, lng);
} 
lat, lng 파싱 로직이 여러 곳에서 반복되고 있어서,
유지보수성과 재사용성을 고려해 parseLatLng() 같은 헬퍼 함수로 분리해보는 건 어떨까요?
다양한 키(lat/latitude, lng/longitude)도 처리할 수 있도록 하면 안정성도 올라갈 것 같아요.

jinju9513 committed:
refactor: 중복된 위도/경도 파싱 로직을 latlng_helper 함수로 통합하여 코드 간결성과 재사용성 개선

mJayxx commented:
전체적으로 메서드 이름이 직관적이어서 흐름을 파악하기 좋았습니다. 레포지토리, 뷰모델의 역할 구분도 잘 되었고 프로바이더도 파일로 구분되어서 깔끔하네요!

jinsung99123 commented on lib/viewmodels/map_view_model.dart:
```diff
@@ -0,0 +1,92 @@
+      southwest: LatLng(southwestLat, southwestLng),
+      northeast: LatLng(northeastLat, northeastLng),
+    );
+  }
```
단순 좌표 변환은 lating helper로 옮겨도 될 것 같습니다.

jinsung99123 commented on lib/views/location/map_page.dart:
```diff
@@ -1,96 +1,216 @@
 
-class MapPage extends StatefulWidget {
-  const MapPage({super.key});
+class MapPage extends ConsumerStatefulWidget {
```
ui 최상위 페이지가 200줄이 넘어가기에 분리를 하면 가독성이 좋아질 것 같습니다!

jinsung99123 commented on lib/views/location/map_page.dart:
```diff
@@ -1,96 +1,216 @@
+        CameraUpdate.newLatLngZoom(latLngs.first, 14),
+      );
+    }
+  }
```
뷰모델로 빼셔도 될 것 같습니다

jinsung99123 commented on lib/views/location/map_page.dart:
```diff
@@ -1,96 +1,216 @@
+              );
+            })
+            .whereType<Marker>()
+            .toSet();
```
유틸에 map_utils 따로 생성후 빼셔도 될 것 같습니다.

jinju9513 committed:
[Location] kakao검색 결과 기반 google map 마커 띄우기
Fixes #56
refector : MapPage 리팩토링
-카메라 위치 초기화, 마커 관리 유틸 적용
-TabController 및 PageController 초기화 로직 분리 및 안전성 개선
-카메라 위치 초기화 함수 분리로 빌드 메서드 가독성 향상
-마커 생성 및 좌표 파싱 유틸 함수로 분리하여 재사용성 증가
-로딩 상태 체크 및 예외 처리 보완
